fvtest/compilerunittest/abstractinterpreter/AbsInterpreterTest.cpp

```diff
@@ -0,0 +1,443 @@
+    stack2.push(&value5);
+    stack2.push(&value6);
+
+    stack1.merge(&stack2, region());
+
+    // merged stack size should not be changed.
+    ASSERT_EQ(3, stack1.size());
+
+    // other should not be affected
+    ASSERT_EQ(3, stack2.size());
+    ASSERT_EQ(&value6, stack2.pop());
+    ASSERT_EQ(&value5, stack2.pop());
+    ASSERT_EQ(&value4, stack2.pop());
+
+    TRTest::AbsTestValue* v3 = static_cast<TRTest::AbsTestValue*>(stack1.pop());
+    ASSERT_EQ(INT_MIN, v3->getLow());
+    ASSERT_EQ(INT_MAX, v3->getHigh());
+    ASSERT_TRUE(v3->isTop());
+
+    TRTest::AbsTestValue* v2 = static_cast<TRTest::AbsTestValue*>(stack1.pop());
+    ASSERT_EQ(-5, v2->getLow());
+    ASSERT_EQ(7, v2->getHigh());
+    ASSERT_FALSE(v2->isTop());
+
+    TRTest::AbsTestValue* v1 = static_cast<TRTest::AbsTestValue*>(stack1.pop());
+    ASSERT_EQ(0, v1->getLow());
+    ASSERT_EQ(1, v1->getHigh());
+    ASSERT_FALSE(v1->isTop());
+}
+
+
+class AbsOpArrayTest : public TRTest::AbsInterpreterTest {};
+
+TEST_F(AbsOpArrayTest, testSize) {
+    TR::AbsOpArray array(3, region());
+    ASSERT_EQ(3, array.size());
+
+    TR::AbsOpArray array2(5, region());
+    ASSERT_EQ(5, array2.size());
+}
+
+TEST_F(AbsOpArrayTest, testSetAndAt) {
+    TR::AbsOpArray array(3, region());
+
+    TRTest::AbsTestValue value1(TR::Int32, 0, 0);
+    TRTest::AbsTestValue value2(TR::Int32, 0, 0);
+    TRTest::AbsTestValue value3(TR::Int32, 0, 0);
+
+    array.set(0, &value1);
+    array.set(1, &value2);
+    array.set(2, &value3);
+    ASSERT_EQ(&value1, array.at(0));
+    ASSERT_EQ(&value2, array.at(1));
+    ASSERT_EQ(&value3, array.at(2));
+}
+
+TEST_F(AbsOpArrayTest, testCloneOperation) {
+    TR::AbsOpArray array(3, region());
+
+    TRTest::AbsTestValue value1(TR::Int32, 0, 0);
+    TRTest::AbsTestValue value2(TR::Int32, 0, 0);
+    TRTest::AbsTestValue value3(TR::Int32, 0, 0);
+
+    array.set(0, &value1);
+    array.set(1, &value2);
+    array.set(2, &value3);
+
+    TR::AbsOpArray* array2 = array.clone(region());
+
+    // cloned stack should have a different pointer than the original
+    ASSERT_NE(array2, &array);
+
+    // the array being cloned should not be affected.
+    ASSERT_EQ(3, array.size());
+    ASSERT_EQ(&value1, array.at(0));
+    ASSERT_EQ(&value2, array.at(1));
+    ASSERT_EQ(&value3, array.at(2));
+
+    // cloned array should have different abstract value pointers.
+    ASSERT_EQ(3, array2->size());
+    ASSERT_NE(&value1, array2->at(0));
+    ASSERT_NE(&value2, array2->at(1));
+    ASSERT_NE(&value3, array2->at(2));
+}
+
+TEST_F(AbsOpArrayTest, testMergeOperation) {
+    TR::AbsOpArray array(3, region());
+
+    TRTest::AbsTestValue value1(TR::Int32, 1, 1);
+    TRTest::AbsTestValue value2(TR::Int32, 5, 8);
+    TRTest::AbsTestValue value3(TR::Int32, INT_MIN, INT_MAX);
+
+    array.set(0, &value1);
+    array.set(1, &value2);
+    array.set(2, &value3);
+
+    TR::AbsOpArray array2(3, region());
+
+    TRTest::AbsTestValue value4(TR::Int32, -5, -10);
```

Comment:
Uhm... does this actually make sense? The high value here is `-10` which is _less than_ `-5`.

Reply:
Thanks for identifying it. It is a mocking abs value so it does not pose any restrictions on the values of low and high. I will
 fix that.